### DISCUSSION TITLE
Extensions to ingredient fuzzy search

### DISCUSSION THREAD

**Discussion by apoorvkh** [mealie-recipes/mealie#950] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
If you want to prepare a recipe immediately (with only ingredients you already have), it's not efficient to use the fuzzy search as it's currently implemented. Each time this happens, a user would have to go through the fuzzy search filter click through and click through each ingredient (and there are many).

**Suggest A Solution**
You can have a "Pantry" tab where you present parsed ingredients to the user (sorted by frequency, as they appear in recipes) and the user can quickly check off the ingredients they have (or could likely substitute). This is saved and the user can later update this list. Then, when filtering recipes, you can offer an option to "show only recipes with available ingredients".
One drawback is that it might still take some time for the user to check all ingredients. Maybe you could cluster ingredients by category (e.g. pantry staples, fridge staples, proteins, vegetables, cuisine staples, etc) and allow the user to "select all" for each category (which might make the process faster for certain categories that are often mostly stocked.
This feature could be in addition to the existing fuzzy search. So a user could check "show only recipes with available ingredients" and then mark specific ingredients they would like to use in this particular query.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by Sponsoredby** [comment]
agreed biggest thing missing is a pantry functionality
